.github/workflows/integration_tests.yml

```diff
@@ -384,18 +347,403 @@ jobs:
               additional_flags+=(--cmake_flag=-DFIREBASE_USE_BORINGSSL=ON)
             fi
           fi
-          python scripts/gha/build_testapps.py --t ${{ needs.prepare_matrix.outputs.apis }} --p ${{ matrix.target_platform }} --output_directory "${{ github.workspace }}" --ios_sdk "${{ env.mobileTestOn }}" --noadd_timestamp ${additional_flags[*]} --short_output_paths
+          python scripts/gha/build_testapps.py --t ${{ needs.prepare_matrix.outputs.apis }} --p Desktop --output_directory "${{ github.workspace }}" --noadd_timestamp ${additional_flags[*]} --short_output_paths --artifact_name "desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}"
 
-      - name: Run desktop integration tests
-        if: matrix.target_platform == 'Desktop' && !cancelled()
+      - name: Upload Desktop integration tests artifact
+        uses: actions/upload-artifact@v2.2.2
+        if: ${{ !cancelled() }}
+        with:
+          name: testapps-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}.zip
+          path: testapps-desktop-${{ matrix.os }}-${{ matrix.ssl_variant }}.zip
+          retention-days: 2
+      - name: Upload Desktop build results artifact
+        uses: actions/upload-artifact@v2.2.2
+        if: ${{ !cancelled() }}
+        with:
+          name: log-artifact
```

Comment:
Shouldn't the artifact name have the matrix fields in it? Same otherwise.

Reply:
Since we never download one single log file. I believe it's good to put them in the same place "log-artifact".